compute_sdk/tests/integration/test_serialization.py

```diff
@@ -382,3 +398,139 @@ def test_check_strategies(strategy_code, strategy_data, function, args, kwargs):
     new_result = new_fn(*new_args, **new_kwargs)
 
     assert original_result == new_result
+
+
+@pytest.mark.parametrize("disallowed_strategy", concretes.SELECTABLE_STRATEGIES)
+def test_allowed_deserializers(disallowed_strategy):
+    allowlist = [
+        strategy
+        for strategy in concretes.SELECTABLE_STRATEGIES
+        if strategy.for_code == disallowed_strategy.for_code
+        and strategy != disallowed_strategy
+    ]
```

Comment:
Consider reducing the in-test complexity.  One thought would be to make two tests via the parameterization:
```python
@pytest.mark.parametrize("disallowed", (s for s in concretes.SELECTABLE_STRATEGIES if s.for_code))
def test_allowed_deserializers_code(...):
    ...

@pytest.mark.parametrize("disallowed", (s for s in concretes.SELECTABLE_STRATEGIES if not s.for_code))
def test_allowed_deserializers_data(...):
    ...

Reply:
split it in two - does that look better?